core/libs/commonwealth/commonwealth/mavlink_comm/MavlinkComm.py

```diff
@@ -51,28 +68,58 @@ async def get_mavlink_message(
                 async with session.get(request_url, timeout=request_timeout) as response:
                     if not response.status == 200:
                         raise MavlinkMessageReceiveFail(f"Received status code of {response.status}.")
+                    # if message is "None", try re-detecting systemid
+                    if await response.text() == "None":
+                        self.set_system_id(await self.get_most_recent_vehicle_id())
+                        raise MavlinkMessageReceiveFail("Received empty response")
                     message = await response.json()
             except asyncio.exceptions.TimeoutError as error:
                 raise MavlinkMessageReceiveFail(f"Request timed out after {request_timeout} second.") from error
 
         return message
 
+    async def get_most_recent_vehicle_id(self) -> int:
+        json_data = await self.get_all_mavlink()
+        most_recent_timestamp = datetime.min.replace(tzinfo=timezone.utc)
+        most_recent_vehicle_id = None
+        for vehicle_id, vehicle in json_data["vehicles"].items():
+            # skip vehicles with id 2 (mavlink-router?) and 255 (common GCS)
```

Comment:
What ? How mavlink-router is using system-id 2 ?

Reply:
honestly I'm not sure of who this is, but since logging is involved, I assumed mavlink-router:
<img width="608" alt="image" src="https://user-images.githubusercontent.com/4013804/230605449-7102a310-cb6a-4001-badb-3ea935e589b4.png">

Reply:
since the code is checking for vehicle type and heartbeat messages, that code was redundant, so I removed it.
I'd still like to know who ID 2 is, though

Reply:
https://github.com/mavlink-router/mavlink-router/blob/3b48da112545e1683c776c6131ff410425e8a978/src/binlog.cpp#L35
https://github.com/mavlink-router/mavlink-router/blob/3b48da112545e1683c776c6131ff410425e8a978/src/logendpoint.h#L30

![](https://media.tenor.com/ZFc20z8DItkAAAAM/facepalm-really.gif)

Reply:
Please add a comment explaining the findings

Reply:
I removed the code, but yeah 🤦

Reply:
Ref: https://github.com/mavlink-router/mavlink-router/issues/392